common/cache/cache_test.go

```diff
@@ -0,0 +1,133 @@
+package cache
+
+import (
+	"testing"
+)
+
+func TestNewLRUCache(t *testing.T) {
+	lruCache := New(5)
+	lruCache.Add("hello", "world")
+	c := lruCache.Contains("hello")
+	if !c {
+		t.Fatal("expected cache to contain \"hello\" key")
+	}
+
+	v := lruCache.Get("hello")
+	if v == nil {
+		t.Fatal("expected cache to contain \"hello\" key")
+	}
+	if v.(string) != "world" {
+		t.Fatal("expected \"hello\" key to contain value \"world\"")
+	}
+
+	r := lruCache.Remove("hello")
+	if !r {
+		t.Fatal("expected \"hello\" key to be removed from cache")
+	}
+
+	v = lruCache.Get("hello")
+	if v != nil {
+		t.Fatal("expected cache to not contain \"hello\" key")
+	}
+}
+
+func TestNewLRUCache_ContainsOrAdd(t *testing.T) {
+	lruCache := New(5)
+
+	f := lruCache.ContainsOrAdd("hello", "world")
+	if f {
+		t.Fatal("expected ContainsOrAdd() to add new key when not found")
+	}
+
+	f = lruCache.ContainsOrAdd("hello", "world")
+	if !f {
+		t.Fatal("expected ContainsOrAdd() to return true when key found")
+	}
+}
+
+func TestNewLRUCache_Clear(t *testing.T) {
+	lruCache := New(5)
+	for x := 0; x < 5; x++ {
+		lruCache.Add(x, x)
+	}
+	if lruCache.Len() != 5 {
+		t.Fatal("expected cache to have 5 entries")
+	}
+	lruCache.Clear()
+	if lruCache.Len() != 0 {
+		t.Fatal("expected cache to have 0 entries")
+	}
+}
+
+func TestLRUCache_Add(t *testing.T) {
+	lruCache := New(2)
+	lruCache.Add(1, 1)
+	lruCache.Add(2, 2)
+	if lruCache.Len() != 2 {
+		t.Fatal("expected cache to have 2 entries")
+	}
+	lruCache.Add(3, 3)
+	if lruCache.Len() != 2 {
+		t.Fatal("expected cache to have 2 entries")
+	}
+
+	v := lruCache.Get(1)
+	if v != nil {
+		t.Fatal("expected cache to no longer contain \"1\" key")
+	}
+	v = lruCache.Get(2)
```

Comment:
Could you add a check here that `Get(2)` brings 2 to the newest?

Reply:
Done